File: python/integrations/openai/gpt_api.py
import os
import httpx

class GptApi:
    def __init__(self, openai_api_key):
        self.api_key = openai_api_key
        self.endpoint = "https://api.openai.com/v1/chat/completions"

    async def generate_text_async(self, prompt: str, model: str = "gpt-3.5-turbo", max_tokens: int = 50, temperature: float = 1.0):
        headers = {
            "Content-Type": "application/json",
            "Authorization": f"Bearer {self.api_key}"
        }

        data = {
            "model": model,
            "messages": [
                {
                    "role": "system",
                    "content": "You are a helpful assistant."
                },
                {
                    "role": "user",
                    "content": prompt
                }
            ],
            "max_tokens": max_tokens,
            "temperature": temperature
        }

        async with httpx.AsyncClient() as client:
            response = await client.post(self.endpoint, json=data, headers=headers)

        if response.status_code != 200:
            raise Exception(f"Failed to generate text: {response.text}")

        result = response.json()
        generated_text = result['choices'][0]['message']['content']
        return generated_text.strip()

File: StableDiffusionTools.Integrations.OpenAi/GptApi.cs
using OpenAI.GPT3;
using OpenAI.GPT3.Interfaces;
using OpenAI.GPT3.Managers;
using OpenAI.GPT3.ObjectModels;
using OpenAI.GPT3.ObjectModels.RequestModels;

namespace StableDiffusionTools.Integrations.OpenAi;

public class GptApi
{
    private readonly IOpenAIService _openAiService;

    public GptApi(OpenAIService openAiService)
    {
        _openAiService = openAiService;
    }

    public async Task<string> GenerateTextAsync(string prompt)
    {
        var request = new ChatCompletionCreateRequest
        {
            Model = Models.ChatGpt3_5Turbo,
            Messages = new List<ChatMessage>
            {
                ChatMessage.FromUser(prompt)
            },
            Temperature = 0.01f
        };

        var response = await _openAiService.ChatCompletion.CreateCompletion(request);

        if (response.Successful)
        {
            return response.Choices.First().Message.Content;
        }

        throw new Exception($"Failed to generate text: {response.Error?.Message}");
    }
}